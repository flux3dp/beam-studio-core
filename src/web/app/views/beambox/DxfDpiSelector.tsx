import * as i18n from 'helpers/i18n';
import * as React from 'react';
import ButtonGroup from 'app/widgets/Button-Group';
import KeyCodeConstants from 'app/constants/keycode-constants';

const lang = i18n.lang;

interface Props {
  defaultDpiValue: number;
  onSubmit: (dpi: number) => void;
  onCancel: () => void;
}

const DxfDpiSelector = ({ defaultDpiValue, onSubmit, onCancel }: Props) => {
    const submitValue = () => {
        const dpi = Number($('#dpi-input').val());
        onSubmit(dpi);
    };
    const _handleKeyDown = (e) => {
        if (e.keyCode === KeyCodeConstants.KEY_RETURN) {
            submitValue();
        }
    };
    const clearInputValue = () => {
        $('#dpi-input').val('');
    };

    const buttons = [
        {
            key: 'cancel',
            label: lang.alert.cancel,
            right: true,
            onClick: () => onCancel()
        },
        {
            key: 'ok',
            className: 'btn-default primary',
            label: lang.alert.ok,
            right: true,
            onClick: () => submitValue()
        }
    ];
    const style = {
        padding: '3px 10px',
        width: '120px',
        'text-align': 'left',
    };
    return (
        <div className='dxf-dpi-selector'>
            <div className='caption'>
                {lang.message.please_enter_dpi}
                <br/>
                1, 2.54, 25.4, 72, 96 etc.
            </div>
            <div className="message" style={{textAlign: 'center'}}>
                <input
                    id='dpi-input'
                    defaultValue={defaultDpiValue}
                    onClick={clearInputValue}
                    onKeyDown={_handleKeyDown}
                    style={style}
                />
            </div>

            <ButtonGroup buttons={buttons}/>
        </div>
    );
}

export default DxfDpiSelector;
