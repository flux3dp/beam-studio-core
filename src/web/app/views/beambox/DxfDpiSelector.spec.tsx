/* eslint-disable import/first */
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DxfDpiSelector from './DxfDpiSelector';

jest.mock('helpers/i18n', () => ({
  lang: {
    alert: {
      cancel: 'Cancel',
      ok: 'OK',
    },
    message: {
      please_enter_dpi: 'Please enter the Unit of your file',
    },
  },
}));

describe('should render correctly', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  test('submit through input', () => {
    document.body.innerHTML = `
      <input id="dpi-input" value="25.4" >
    `;
    const onSubmit = jest.fn();
    const onCancel = jest.fn();
    const wrapper = shallow(<DxfDpiSelector
      defaultDpiValue={100}
      onSubmit={onSubmit}
      onCancel={onCancel}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('input#dpi-input').simulate('keydown', {
      keyCode: 14,
    });
    expect(onSubmit).not.toHaveBeenCalled();

    wrapper.find('input#dpi-input').simulate('keydown', {
      keyCode: 13,
    });
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenNthCalledWith(1, 25.4);

    wrapper.find('input#dpi-input').simulate('click');
    expect($('#dpi-input').val()).toBe('');
  });

  test('submit through buttons', () => {
    document.body.innerHTML = `
    <select id="dpi-input">
      <option>1</option>
      <option>2.54</option>
      <option selected="selected">25.4</option>
      <option>72</option>
      <option>96</option>
    </select>
  `;
    const onSubmit = jest.fn();
    const onCancel = jest.fn();
    const wrapper = shallow(<DxfDpiSelector
      defaultDpiValue={100}
      onSubmit={onSubmit}
      onCancel={onCancel}
    />);

    wrapper.find('ButtonGroup').props().buttons[1].onClick();
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenNthCalledWith(1, 25.4);

    wrapper.find('ButtonGroup').props().buttons[0].onClick();
    expect(onCancel).toHaveBeenCalledTimes(1);
  });
});
