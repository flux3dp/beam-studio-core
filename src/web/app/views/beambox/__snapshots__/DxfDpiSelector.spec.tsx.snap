// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly submit through input 1`] = `
<div
  className="dxf-dpi-selector"
>
  <div
    className="caption"
  >
    Please enter the Unit of your file
    <br />
    1, 2.54, 25.4, 72, 96 etc.
  </div>
  <div
    className="message"
    style={
      {
        "textAlign": "center",
      }
    }
  >
    <input
      defaultValue={100}
      id="dpi-input"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        {
          "padding": "3px 10px",
          "textAlign": "left",
          "width": "120px",
        }
      }
    />
  </div>
  <ButtonGroup
    buttons={
      [
        {
          "key": "cancel",
          "label": "Cancel",
          "onClick": [Function],
          "right": true,
        },
        {
          "className": "btn-default primary",
          "key": "ok",
          "label": "OK",
          "onClick": [Function],
          "right": true,
        },
      ]
    }
  />
</div>
`;
